server/paymentProviders/stripe/virtual-cards.js

```diff
@@ -52,23 +57,111 @@ export const assignCardToCollective = async (cardNumber, expireDate, cvv, collec
   return await models.VirtualCard.create(cardData);
 };
 
+export const processAuthorization = async (stripeAuthorization, stripeEvent) => {
+  const virtualCard = await getVirtualCardForTransaction(stripeAuthorization.card.id);
+
+  if (!virtualCard) {
+    throw new Error(`Virtual card ${stripeAuthorization.card.id} not found`);
+  }
+
+  const host = virtualCard.host;
+
+  await checkStripeEvent(host, stripeEvent);
+
+  const amount = stripeAuthorization.amount;
```

Comment:
Would be great to handle zero-decimal currencies properly here. https://stripe.com/docs/currencies#zero-decimal. Search in the codebase convertToStripeAmount ZERO_DECIMAL_CURRENCIES

Reply:
To make it future proof, some currency conversion could be interesting, converting balance or amount to the other currency.

Reply:
Both suggestions can be added as TODO notes.